articles/architecture-scenarios/b2c/user-provisioning.md

```diff
@@ -0,0 +1,83 @@
+2. If you have legacy user data, do you want to migrate it to Auth0?
+3. Do you want your users to sign up with their existing accounts like Facebook or Google? 
+
+## Identity data storage
+
+During sign up, Auth0 creates a [user profile](/users/concepts/overview-user-profile) that contains information about the user’s account. You can choose to store the user information in Auth0's data store or your own. 
+
+### Use your own identity data store
+
+If you want to use your own identity store, perhaps because you’ve got applications which you aren’t ready to migrate or can’t be migrated, use Auth0’s [Database Connections](/connections/database/custom-db) to proxy an existing identity store. 
+
+### Use Auth0 as the identity data store
+
+Auth0 provides identity data storage to help you manage the burden of storing user credentials safely and securely. If you have a legacy identity store that you would like Auth0 to host, you can use Auth0’s user migration capabilities. 
+
+## User migration
+
+If you decide to use Auth0 instead of your own identity store, you can [migrate users](/users/concepts/overview-user-migration) with bulk migration or progressively with automatic migration. You can also choose a two-phased approach to user migration, using automatic migration first, then using bulk migration for the users that remain. For other scenarios, see [User Migration Scenarios](/users/references/user-migration-scenarios).
+
+### Bulk migration
+
+Auth0 recommends using the [Management API](/users/concepts/overview-user-migration#bulk-user-imports-with-the-management-api) for bulk migration over the [User Import/Export extension](https://auth0.com/docs/users/concepts/overview-user-migration#migrate-users-with-the-user-import-export-extension) in all but the most simple cases. The Management API provides more flexibility and control. 
+
+After the bulk migration is complete, your users will need to reset their passwords.
+
+### Automatic migration
+
+[Automatic migration](/users/guides/configure-automatic-migration) allows users to be migrated incrementally, so they will not need to reset their passowrds. 
+
+## Self Sign Up
+
+Self sign up leverages Auth0 database connections to store the user id, password and username collected during the sign up process. Using the Auth0 Dashboard, you can configure the database connection policies such as minimum username length and password options such as password strength and complexity.  
+
+Auth0 provides a widget, [Lock](/libraries), which provides user interface functionality for sign up out-of-the-box. Lock supports a range of customization for user self sign up as well as login, however if you need to make extensive changes to CSS, Javascript or HTML to achieve your branding goals then you should use a fully customized [Universal Login](/universal-login) instead. For a comparison of the capabilities, see [Lock vs. a Custom UI](/libraries/when-to-use-lock).
+
```

Comment:
Auth0 offers [Lock](/libraries), which providers a user interface that includes signup functionality out of the box. Lock supports a some customization to its appearance, but if you would like to make extensive changes to the login widget to achieve your branding goals, we recommend that you use [Universal Login](/universal-login) instead.  For a comparison of the capabilities, see [Lock vs. a Custom UI](/libraries/when-to-use-lock).

Reply:
This comparison isn't really right. If you want to make extensive changes, you would go from Lock to Auth0.js.

Universal Login isn't the custom alternative to Lock.

However, we really should be proffering Universal Login first, and noting that if it doesn't meet your needs, we also have libraries that can be embedded in your application, such as Lock (premade login widget) and Auth0.js (a library with which you can build your own login UI).